profiles/views.py

```diff
@@ -36,4 +38,26 @@ def edit(request):
     else:
         form = EditProfileForm(initial=initial)
 
-    return render(request, "profile/edit.html", {"form": form}, status=status)
\ No newline at end of file
+    return render(request, "profile/edit.html", {"form": form}, status=status)
+
+
+def follow(request, profile):
+    target_user = get_object_or_404(get_user_model(), username=profile)
+    if request.user.is_authenticated:
+        is_following = request.user.profile.is_following(target_user.profile)
+    else:
+        is_following = False
+
+    context = {
+        "target_user": target_user,
+        "is_following": is_following,
+    }
+
+    if request.method == "GET":
+        return render(request, "profile/_follow.html", context)
+    elif request.method == "POST":
+        if is_following:
+            request.user.profile.unfollow(target_user.profile)
+        else:
+            request.user.profile.follow(target_user.profile)
+        return HttpResponseRedirect(reverse('follow_profile', kwargs={"profile": profile}), status=303)
```

Comment:
Nice! I might add another `request.user.is_authenticated` for the POST though:

```python
def follow(request, profile):
    target_user = get_object_or_404(get_user_model(), username=profile)
    if request.user.is_authenticated:
        is_following = request.user.profile.is_following(target_user.profile)
    else:
        is_following = False

    context = {
        "target_user": target_user,
        "is_following": is_following,
    }

    if request.method == "POST" and request.user.is_authenticated:
        if is_following:
            request.user.profile.unfollow(target_user.profile)
        else:
            request.user.profile.follow(target_user.profile)
        return HttpResponseRedirect(reverse('follow_profile', kwargs={"profile": profile}), status=303)
    
    return render(request, "profile/_follow.html", context)
```

Reply:
Do we need the 303 in a frame?

Reply:
303 isn't strictly necessary, we could just render the form again, or just return a default 302. For consistency's sake, though, I kind of like the 303. But it is an extra request compared to just rendering again.